### PR TITLE
Add shared connection code for mysql modules

### PR DESCRIPTION
**Pull request by Jmainguy**

### CODE REVIEW

**Comment by Jmainguy** [comment]
This needs to be merged before https://github.com/ansible/ansible-modules-core/pull/2447 can be. This allows all 3 to use the same connection method, which greatly simplifies the modules.

File: lib/ansible/module_utils/mysql.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# This code is part of Ansible, but is an independent component.
```
Comment (bcoca) [inline lib/ansible/module_utils/mysql.py:1]: module utils code does not use the GPL but a BSD license, this is so others can reuse this code when distributing independent 3rd party modules with any license they like.

**Comment by bcoca** [comment]
license issue aside, looks good to me.

note: i would also add a mysql document fragment and add the connection options to it so we can have the documentation also in only 1 place and not duplicated across modules.  That is not a requirement for merging this PR in.

**Comment by abadger** [comment]
Okay, merged. this refactor :-)

**Pull request merged**